### ISSUE TITLE
Protocol 11 Compatibility

### ISSUE THREAD

**Issue by theaeolianmachine** [astroband/ruby-stellar-sdk#55]
Protocol V11 of Stellar adds a new `manage_buy_offer` operation. Also names of `manage_offer` and `create_passive_offer` have been changed to `manage_sell_offer` and `create_passive_sell_offer`.

References:
* https://github.com/stellar/go/pull/1164
* https://github.com/stellar/stellar-protocol/blob/master/core/cap-0006.md
* [XDR files diff](https://github.com/stellar/stellar-core/compare/v10.3.0...v11.0.0#diff-01cb6b2520b4d5097b53837c1c18c5a3) - this shows all changes between v10.3.0 and v11.0.0 in Stellar Core — you'll need to just look at the changes in `src/xdr` (realistically any file that ends with a `.x` extension).

Let me know if you have any questions!

**Comment by theaeolianmachine** [comment]
@ramontayag, wasn't able to place issues on the fork — have you disabled this, or is this because it's a fork? Is there anything we can do to make the workflow easier for you?

**Comment by ramontayag** [comment]
@theaeolianmachine looks like it's off by default for forks. I just enabled Issues on it. :)

I'm not familiar with the XDR process. I've seen commits here, but do you know where I can find info to regenerate XDR files so I can import it?

**Comment by theaeolianmachine** [comment]
Awesome — I actually talked to the platform team, and think we should do the following:
- Delete your fork — especially given the number of PRs and issues, it'll be easy to migrate
- Create a new repository under bloom-solutions that has its own issues and PRs
- We'll update the docs website to point to your SDK, make a larger notice on our repo that points to yours, and I'll help migrate any issues currently in this repository.

If you could do that and point me the new repo I'll get the ball rolling.

As for the XDR process, you'll need to look in the [src/xdr folder in stellar-core](https://github.com/stellar/stellar-core/tree/v11.0.0/src/xdr) at the v11.0.0 tag, and then use [xdrgen](https://github.com/stellar/xdrgen) to create the new xdr files. From there it's just making sure everything works and is up to date.

**Comment by nebolsin** [comment]
Just a quick note, we have a dedicated rake task in ruby-stellar-base to update XDR, so ideally `rake xdr:update` should bring generated code in sync with the latest XDR definitions from stellar-core.

**Comment by nebolsin** [comment]
@theaeolianmachine Regarding the migration, I think the simplest way is to delete the forks and then transfer ownership of `stellar/ruby-stellar-base` and `stellar/ruby-stellar-sdk` to bloom-solutions. This way we’ll preserve all the issues/PR history and also Github will put the proper redirects from the old repo locations, so that both web and bundler/git links will continue to work and allow smooth transition.

**Comment by ire-and-curses** [comment]
Here are docs about transferring ownership: https://help.github.com/en/articles/transferring-a-repository

**Comment by theaeolianmachine** [comment]
Thanks for mentioning @nebolsin, didn't know that was an option. @ramontayag can you open a pull request and make sure (haven't looked yet personally) that our master repo is up to date with your fork?

Once you've done that, can you delete it? @tomquisel, I'll need you to initiate the transfer once this is complete, por favor.

**Comment by tomquisel** [comment]
Good call on transferring the repo. It sounds like these are the steps:

- [x] @ramontayag submits PR to `stellar/ruby-stellar-sdk` `stellar/ruby-stellar-base` to bring them up to date with his forks
- [x] @theaeolianmachine merges those PRs
- [x] @ramontayag deletes his forks
- [x] @tomquisel transfers both repos to @ramontayag 

Just ping me when I'm needed for the last step!

**Comment by nebolsin** [comment]
@ramontayag @theaeolianmachine I was in need of v11 support for one of my tasks, so I went ahead and created stellar/ruby-stellar-base#49 to incorporate the changes from a fork and stellar/ruby-stellar-base#48 which builds on top of it and brings support for protocol 11.

**Comment by ramontayag** [comment]
@tomquisel ok done the first task!

**Comment by ramontayag** [comment]
@nebolsin ok cool. I added friendbot support in our fork. I haven't worked on v11 yet so I would appreciate your help there 👍

**Comment by nebolsin** [comment]
@ramontayag My PRs are for ruby-stellar-base, not ruby-stellar-sdk, so they don't interfere with your friendbot changes. PR with v11 support is ready for review in `ruby-stellar-base` (see above), and I don't think v11 requires any changes on `ruby-stellar-sdk` side, so I'm not sure how else can I help here.

**Comment by tomquisel** [comment]
@ramontayag I've merged your PRs, we're all ready for you to delete your forks and then we'll transfer these repos over.

**Comment by theaeolianmachine** [comment]
Hey @tomquisel I just took a look, and both forks have been deleted. Let's get both of these repos transferred over (all the PRs have been merged as well for recent updates for both this and https://github.com/stellar/ruby-stellar-base)

**Comment by tomquisel** [comment]
Great! It's moved over to @ramontayag , now he just needs to move it to `bloom-solutions`

**Comment by theaeolianmachine** [comment]
Alright everything's been moved over, and it looks like @nebolsin's PRs are merged. If everything is good to go here, please close this out, and I'd appreciate it if anyone lets me know when a release is cut for V11!

**Comment by ramontayag** [comment]
@nebolsin are you able to update XDR? I ran the rake task you mentioned, but I have OTP enabled and will have to find a solution to it -- like allowing users to enter their OTP when calling the rake task. For now, it might be easier for you to update it since there are people waiting on this.

**Comment by nebolsin** [comment]
@ramontayag I believe my PR with full protocol 11 support has already been merged in ruby-stellar-base (including the updated XDR), so I’m not sure what are you asking about...

**Comment by ramontayag** [comment]
Ah ok, this was probably merged by SDF. I think I just need to cut a release then? Thanks

**Comment by ramontayag** [comment]
Released stellar-base v0.20.0 :+1:

**Comment by nebolsin** [comment]
@ramontayag Yes, I think so. Regarding OTP, if you're talking about 2fa on your Github account, rake task only uses Github API to read from a public `stellar/stellar-core` repo, so authorization shouldn't be necessary. But if it is, the rake task fully supports `.netrc` file, so you can [generate an API token for the command line](https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line) and put the following section into your `~/.netrc`:

```
machine api.github.com
login ramontayag
password <your token>
```
